src/scripts/patch_ingame.lua

```diff
@@ -18,23 +18,78 @@ ScriptCB_DoFile("utility_functions2")
 --ReadDataFile("v1.3patch_strings.lvl") -- where to put, root of'0'?
 
+-- trim "path\\to\\file.lvl" to "file"
+function trimToFileName(filePath)
+    local separatorIndex = 0
+    local lastSeparatorIndex = 0
 
+    -- Find the last occurrence of '\\'
+    local i = 1
+    while true do
+        local nextIndex = string.find(filePath, "\\", i, true)
+        if nextIndex then
+            lastSeparatorIndex = separatorIndex
+            separatorIndex = nextIndex
+            i = nextIndex + 1
+        else
+            break
+        end
+    end
+
+    local extensionIndex = string.find(filePath, "%.[^.]+$") -- Find the last occurrence of '.' followed by characters not including '.'
+    if extensionIndex then
+        filePath = string.sub(filePath, 1, extensionIndex - 1) -- Trim file extension
+    end
+
+    -- Extract substring after the last separator
+    if lastSeparatorIndex > 0 then
+        return string.sub(filePath, lastSeparatorIndex + 1)
+    else
+        return filePath
+    end
+end
 
 function RunUserScripts()
-    local maxScripts = 100
-    local i = nil
-    local patchDir = "..\\..\\addon\\0\\"
+
     print("RunUserScripts() Start")
-    for i = 0, maxScripts, 1 do
-        
-        if ScriptCB_IsFileExist(patchDir .. "user_script_" .. i .. ".lvl") ~= 0 then
-            print("game_interface: Found user_script_" .. i .. ".lvl")
-            
-            ReadDataFile(patchDir .. "user_script_" .. i .. ".lvl")
-            ScriptCB_DoFile("user_script_" .. i)
+
+    -- retrieve user scripts we saved in mission setup table
+    if ScriptCB_IsMissionSetupSaved() then
+        local missionSetup = ScriptCB_LoadMissionSetup()
+        if missionSetup ~= nil then
+            if missionSetup.userScripts ~= nil then
+                for _, scriptPath in missionSetup.userScripts do
+                    if ScriptCB_IsFileExist(scriptPath) then
+                        ReadDataFile(scriptPath)
+                        -- .lvl or .script file must contain a .lua file with matching name
+                        -- that file will be the entrypoint of the user script
+                        ScriptCB_DoFile(trimToFileName(scriptPath))
+                    end
+                end
+
+                local hasOnlyUserScripts = true
+                for key, value in missionSetup do
+                    if key ~= "userScripts" then
+                        hasOnlyUserScripts = false
+                        break
+                    end
+                end
+
+                if hasOnlyUserScripts then
+                    -- if the missionSetup was only user scripts then clear items
+                    ScriptCB_ClearMissionSetup()
+                else
+                    --remove the user scripts and save it again
+                    --we don't really have to do this but i figure its a good idea
+                    missionSetup.userScripts = nil
+                    ScriptCB_SaveMissionSetup(missionSetup)
+                end
```

Comment:
I might remove this part because I'm not sure if you quit back to shell from ingame, do the addon files get loaded again? if they don't then the user script table will be nil

Reply:
Nevermind, when you quit back to shell from a game, it goes through all the addme scripts again